### DISCUSSION TITLE
Running commands after language server is ready

### DISCUSSION THREAD

**Discussion by gi1242** [neoclide/coc.nvim#3901] [Q&A] [answered]
Hi There,

Is it possible to run some code after the language server has initialized? In particular, I'm using `coc-texlab`, and I'd like to run

    CocCommand latex.Build
    CocCommand latex.ForwardSearch

to automaticaly build and start my previewer. But if I do it in my ftplugin files, I get an error as the language server hasn't loaded yet.

Thanks,

GI

**Comment by fannheyward** [comment, marked as answer]
`autocmd User CocNvimInit call CocAction('runCommand', 'latex.Build')`, check `:h CocNvimInit`.

**Reply by chemzqm** [reply]
Need check services state returned by `CocAction('services')`

**Reply by gi1242** [reply]
Thanks so much!
